client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -693,6 +693,9 @@ class LifecycleManager(appId: String, val conf: CelebornConf) extends RpcEndpoin
           failureInfos.add(s"[reserveSlots] Failed to" +
             s" reserve buffers for ${Utils.makeShuffleKey(applicationId, shuffleId)}" +
             s" from worker ${workerInfo.readableAddress()}. Reason: ${res.reason}")
+          if (res.status.equals(StatusCode.WORKER_SHUTDOWN)) {
+            workerStatusTracker.resolveShutdownWorkers(List(workerInfo).asJava)
```

Comment:
Do we need notify the listeners this time？

Reply:
Not so necessary, remove it.